### PR TITLE
testing(api): Adding API equivalency test for celestia-openrpc

### PR DESCRIPTION
**Pull request by distractedm1nd**
Adds a test to ensure API equivalence to celestia-openrpc. It will not be used or added to CI in this repo, hence the build tag. celestia-openrpc will pull this repo, build with the build tag, and run the test itself.

The reason it shouldnt be added to the CI in celestia-node is because it would fail every time we change the API. It shouldn't fail here, it should fail in the other repo.

### CODE REVIEW

File: api/rpc_library_test.go [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+		compareStructs(t, libraryStructs[i], apiStructs[i], moduleNames[i])
+	}
+}
+
+func compareStructs(t *testing.T, structA, structB interface{}, moduleName string) {
+	typeA := reflect.TypeOf(structA)
+	typeB := reflect.TypeOf(structB)
+
+	if typeA.NumField() != typeB.NumField() {
+		t.Fatalf("%s API has a comflicting number of fields: %d != %d", moduleName, typeA.NumField(), typeB.NumField())
```
Comment (ramin) [inline api/rpc_library_test.go:78]: spelling: conflicting

**Review by ramin — COMMENTED** [review]
I think it'd be nice to start to organize these into a top level test folder and intro a more reusable build tag, maybe

`/tests/conformance/api` and the tag being `conformance` then we can run ALL conformance tests, or only a subset by running go test inside `/tests/conformance/<type of test>`

open to another name than conformance

File: api/rpc_library_test.go (hunk 1 above)
Comment (distractedm1nd) [reply api/rpc_library_test.go:78]: thx

**Comment by distractedm1nd** [comment]
I like the name conformance a lot more, I will change the build tag - but I would maybe hold off on moving the test until we have more conformance test cases? WDYT

**Review by renaynay — APPROVED** [review]
utack

**Comment by walldiss** [comment]
I like the idea to have this test. But why test for external client implementation is in node repo?  It seems wrong that node will depend on that client after this PR

**Comment by ramin** [comment]
yeah actually i agree with @walldiss, this should live in the `celestia-openrpc` repo and run as a conformance test there

**Comment by distractedm1nd** [comment]
It can't, that's why its here :( I agree ofc but openrpc can't import node because thats the whole reason it exists

**Comment by renaynay** [comment]
@distractedm1nd make a conformance test repo and put it there.

**Pull request closed without merging**